Sources/Seismography/ParseGIR.swift

```diff
@@ -5,4 +5,302 @@
 //  Created by Robert Widmann on 2/18/18.
 //
 
+import Lithosphere
+import Crust
 import OuterCore
+
+public final class GIRParser {
+  struct ParserScope {
+    let name: String
+  }
+  let parser: Parser
+  var module: GIRModule = GIRModule()
+
+  private var _activeScope: ParserScope?
+  fileprivate var continuationsByName: [String: Continuation] = [:]
+  fileprivate var undefinedContinuation: [Continuation: TokenSyntax] = [:]
+  fileprivate var localValues: [String: Value] = [:]
+  fileprivate var forwardRefLocalValues: [String: TokenSyntax] = [:]
+
+  var currentScope: ParserScope {
+    guard let scope = self._activeScope else {
+      fatalError("Attempted to request current scope without pushing a scope.")
+    }
+    return scope
+  }
+
+  public init(_ parser: Parser) {
+    self.parser = parser
+  }
+
+  func pushScope(named name: String) {
+    self._activeScope = ParserScope(name: name)
+  }
+
+  func popScope() {
+    precondition(self._activeScope != nil)
+    self._activeScope = nil
```

Comment:
This oughta reset locally-defined SSA values.  It needs a test that contains multiple functions.